### PR TITLE
docs: improve build/test docs

### PR DESCRIPTION
**Pull request by akshaysngupta**
## Checklist
- [x] The title of the PR is clear and informative
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] Issues addressed by the PR are mentioned in the description followed by `Fixes`.

## Description

This PR add more documentation on building and testing AGIC code.

### CODE REVIEW

File: docs/developers/build.md [hunk 1]
```diff
@@ -49,11 +40,109 @@ The file will contain a JSON blob with the following shape:
...
+    Version: 1.2.0; Commit: ef716c14; Date: 2020-07-23-18:37T+0000
+    ERROR: logging before flag.Parse: I0723 18:37:31.989656    6766 utils.go:115] Using verbosity level 9 from environment variable APPGW_VERBOSITY_LEVEL
+    ERROR: logging before flag.Parse: I0723 18:37:31.989720    6766 main.go:78] Unable to load cloud provider config ''. Error: Reading Az Context file "" failed: open : no such file or directory
+    E0723 18:37:31.999445    6766 context.go:210] Error fetching AGIC Pod (This may happen if AGIC is running in a test environment). Error: resource name may not be empty
+    I0723 18:37:31.999466    6766 environment.go:240] KUBERNETES_WATCHNAMESPACE is not set. Watching all available namespaces.
+    ...
+    ```
+
+### Visual Studio Code (F5 debugging)
+You can also setup vscode to run the project `F5` and do breakpoint debugging. For this, you need to setup your
```
Comment (3quanfeng) [inline docs/developers/build.md:82]: nit: .. to setup your launch.json file within .vscode folder?

File: docs/developers/build.md [hunk 2]
```diff
@@ -49,11 +40,109 @@ The file will contain a JSON blob with the following shape:
...
+NAME                    CHART VERSION   APP VERSION     DESCRIPTION                                       
+staging/ingress-azure   10486           10486           Use Azure Application Gateway as the ingress fo...
+staging/ingress-azure   10465           10465           Use Azure Application Gateway as the ingress fo...
+staging/ingress-azure   10256           10256           Use Azure Application Gateway as the ingress fo...
+
+# install/upgrade
+helm install ingress-azure \
+      -f helm-config.yaml \
+      application-gateway-kubernetes-ingress/ingress-azure \
+      --version 10486
```
Comment (3quanfeng) [inline docs/developers/build.md:129]: nit: to add a note how to identify the buildId from the url?

File: docs/developers/build.md [hunk 3]
```diff
@@ -49,11 +40,109 @@ The file will contain a JSON blob with the following shape:
...
+```json
+{
+    "version": "0.2.0",
+    "configurations": [
+        {
+            "name": "Debug",
+            "type": "go",
+            "request": "launch",
+            "mode": "debug",
+            "program": "${workspaceFolder}/cmd/appgw-ingress",
```
Comment (3quanfeng) [inline docs/developers/build.md:92]: ${workspaceFolder}/bin/appgw-ingress

File: docs/developers/build.md [hunk 4]
```diff
@@ -1,35 +1,26 @@
...
+    * [Visual Studio Code (F5 debugging)](#visual-studio-code-f5-debugging)
+* [Run on a cluster using a Dev Release](#create-a-dev-release)
+* [CMake options](#cmake-options)
 
 ## Running it locally
 
 This section outlines the environment variables and files necessary to successfully compile and run the Go binary, then connect it to an [Azure Kubernetes Service](https://docs.microsoft.com/en-us/azure/aks/intro-kubernetes).
 
+### Pre-requisite
+* [go version 1.13](https://golang.org/dl/)
```
Comment (JorTurFer) [inline docs/developers/build.md:14]: Is it mandatory go 1.13 or at least go 1.13?

File: docs/developers/build.md [hunk 5]
```diff
@@ -1,35 +1,26 @@
...
-- `dockerize` builds a docker image with the binary from `appgw-ingress` target
-- `dockerpush` pushes the docker image to a container registry with prefix defined in CMake variable `<deployment_push_prefix>`
-
-To run the CMake targets:
-
-1. `mkdir build && cd build` creates and enters a build directory
-2. `cmake ..` generates project configuration in the build directory
-3. `cmake --build .` to build the default target,
-    or `cmake --build . --target <target_name>` to specify a target to run from above
+* [Running it locally](#running-it-locally)
```
Comment (JorTurFer) [inline docs/developers/build.md:3]: Maybe this index should have all the titles in the document

**Pull request merged**